Library/Homebrew/test/utils/svn_spec.rb

```diff
@@ -0,0 +1,40 @@
+require "utils/svn"
+
+describe Utils do
+  describe "#self.svn_available?" do
+    it "returns true if svn --version command succeeds" do
+      allow_any_instance_of(Process::Status).to receive(:success?).and_return(true)
+      expect(described_class.svn_available?).to be_truthy
+    end
+
+    it "returns false if svn --version command does not succeed" do
+      allow_any_instance_of(Process::Status).to receive(:success?).and_return(false)
+      expect(described_class.svn_available?).to be_falsey
+    end
+
+    it "returns svn version if already set" do
+      described_class.instance_variable_set(:@svn, true)
```

Comment:
This will not reset the variable after the test, can you find a way to test the functionality that actually sets this to `true`? Otherwise, the only thing this is testing is whether `svn_available?` uses this variable.